### PR TITLE
DEStats additon from SciMLBase

### PR DESCRIPTION
**Pull request by H-Sax**
This utilise DEStats from SciML base to allow for parallel executions of DDE

### CODE REVIEW

**Review by devmotion — COMMENTED** [review]
I haven't follow the stats updates in SciMLBase closely, but if we should/have to switch to `DEStats` then I assume we can remove the `DDEStats` definitions completely. We should also make sure that the compat entries for SciMLBase are correct (such that `DEStats` is actually available).

File: src/solve.jl [hunk 1]
```diff
@@ -166,7 +166,8 @@ function DiffEqBase.__init(prob::DiffEqBase.AbstractDDEProblem,
         Val(isinplace(prob)))
 
     # separate statistics of the integrator and the history
-    stats = DDEStats(0)
+    #stats = DDEStats(0)
```
[the remark this answers is not in the document]
Comment (H-Sax) [reply src/solve.jl:169]: You are correct I should have provided more context, this change steps out of a conversation on slack 
https://julialang.slack.com/archives/CN04R7WKE/p1708095479463839

**Comment by ChrisRackauckas** [comment]
When https://github.com/JuliaRegistries/General/pull/101462 is merged, bump the lower bound here and I'll merge when tests pass.

**Comment by devmotion** [comment]
Should still remove the DDEStats implementation completely, I guess?

**Pull request merged**